### PR TITLE
support pip mirrors

### PR DESCRIPTION
**Pull request by paulczar**

### CODE REVIEW

**Comment by j2sol** [comment]
LGTM. :mountain_bicyclist:

File: envs/vagrant/mirrors/group_vars/all.yml [hunk 1]
```diff
@@ -46,3 +46,7 @@ apt:
     erlang:
       repo: http://173.247.104.68/packages.erlang-solutions.com/ubuntu
       key_url: http://173.247.104.68/keys/erlang.asc
+
+common:
+  pip:
+    proxy_url: http://173.247.104.68:8092/simple
```
Comment (j2sol) [inline envs/vagrant/mirrors/group_vars/all.yml]: This needs to go into envs/example/ too.

Comment (paulczar) [inline envs/vagrant/mirrors/group_vars/all.yml]: I added an empty common.pip.proxy_url to the defaults for common role. 

checked that pip is okay with that line being empty in the config file and it is.

Comment (j2sol) [inline envs/vagrant/mirrors/group_vars/all.yml]: Yeah, but your default is applying to the final bit of the dictionary, not to the pip dictionary itself, so the tests fail.

File: roles/common/templates/etc/pip.conf [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[global]
+index-url = {{ common.pip.proxy_url|default('') }}
```
Comment (j2sol) [inline roles/common/templates/etc/pip.conf]: I think you'd want to do something more like:
{% if common.pip.proxy_url is defined %}
[global]
index-url = {{ common.pip.proxy_url }}
{% endif %}

**Pull request merged**